src/cobra/util/process_pool.py

```diff
@@ -79,8 +79,14 @@ def __exit__(
         exc_tb: Optional[TracebackType],
     ) -> Optional[bool]:
         """Clean up resources when leaving a context."""
+        # The `multiprocessing.Pool.__exit__` only terminates pool processes. For a
+        # clean exit, we close the pool and join the pool processes first.
+        try:
```

Comment:
Does the code in the `try` block raise exceptions? I'm thinking whether we need the `try ... finally` block at all.

Reply:
I can't remember why I originally put it there (it was in the `close` method). My guess is that I really wanted to clean up the pickle file no matter what. I'm also no longer sure that these changes are needed. 🤷🏼‍♂️

Reply:
I see. 😅